momentsapi/src/main/java/com/tealium/momentsapi/MomentsApiService.kt

```diff
@@ -0,0 +1,104 @@
+package com.tealium.momentsapi
+
+import com.tealium.core.Module
+import com.tealium.core.ModuleFactory
+import com.tealium.core.Modules
+import com.tealium.core.Tealium
+import com.tealium.core.TealiumContext
+import com.tealium.momentsapi.network.HttpClient
+import com.tealium.momentsapi.network.NetworkClient
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.json.JSONObject
+import java.lang.Exception
+import java.net.URL
+
+/**
+ * Interface defining methods for fetching engine responses.
+ */
+interface MomentsApi {
+    /**
+     * Fetches the engine response for the given engine ID.
+     * @param engineId The ID of the engine to fetch the response for.
+     * @param responseListener Listener for handling the response.
+     */
+    fun fetchEngineResponse(engineId: String, responseListener: ResponseListener<EngineResponse>)
+}
+
+/**
+ * Service class providing implementation for the Moments API module.
+ * @param context The Tealium context.
+ * @param networkClient The network client used for making HTTP requests (default is HttpClient).
+ * @param backgroundScope The CoroutineScope used for background tasks (default is Dispatchers.IO).
+ */
+class MomentsApiService @JvmOverloads constructor(
+    private val context: TealiumContext,
+    private val region: MomentsApiRegion,
+    private val networkClient: NetworkClient = HttpClient(context),
+    private val backgroundScope: CoroutineScope = CoroutineScope(Dispatchers.IO)
+) : Module, MomentsApi {
+
+    override var enabled: Boolean = true
+    override val name: String = MODULE_NAME
+
+    private val engineUrl: String =
+        "https://personalization-api.${region.value}.prod.tealiumapis.com/personalization/accounts/${context.config.accountName}/profiles/${context.config.profileName}/"
+
+    private val referer = context.config.momentsApiReferer
+        ?: "https://tags.tiqcdn.com/utag/${context.config.accountName}/${context.config.profileName}/${context.config.environment.environment}/mobile.html"
+
+    override fun fetchEngineResponse(
+        engineId: String,
+        responseListener: ResponseListener<EngineResponse>
+    ) {
+        fetchResponse(engineId, responseListener)
+    }
```

Comment:
Is this just the same method as below just with a different param name? Could we just have 1?